examples/el_key/src/lib.rs

```diff
@@ -0,0 +1,446 @@
+//! An exemple that uses the `el_key` property of virtual DOM elements to define correspondence
+//! betweeen new and old elements.  CSS animation shows how elements are updated, added and
+//! removed.
+
+// Some Clippy linter rules are ignored for the sake of simplicity.
+#![allow(clippy::needless_pass_by_value, clippy::trivially_copy_pass_by_ref)]
+#![allow(
+    clippy::cast_possible_truncation,
+    clippy::cast_possible_wrap,
+    clippy::cast_precision_loss
+)]
```

Comment:
I will disable these warnings only for the functions that clippy complains about.
This code should work if NUM_CARDS constant not greater then 26.
```rust
#[allow(clippy::cast_precision_loss, clippy::cast_possible_truncation)]
fn card_bg_color(card_n: usize, total_cards: usize) -> Srgb {
    let hue = card_n as f32 / total_cards as f32 * 360.0;
    Hsl::new(hue % 360.0, 1.0, 0.7).into()
}

#[allow(clippy::cast_precision_loss, clippy::cast_possible_truncation)]
fn card_fg_color(card_n: usize, total_cards: usize) -> Srgb {
    let hue = card_n as f32 / total_cards as f32 * 360.0;
    Hsl::new((hue + 240.0) % 360.0, 0.8, 0.1).into()
}

#[allow(clippy::cast_possible_truncation)]
const fn card_id(card_n: usize) -> char {
    (b'A' + card_n as u8) as char
}
```

Reply:
26 because of the number of letters in alphabet?
It works for NUM_CARDS <= 32, then compiler fails because it doesn't allow to create arrays longer than 32 items. But even if we change NUM_CARDS type to u8, clippy will be angry because of casting `usize` to `u8` from enumerator into chars..
So the code isn't super clear but it's not the purpose of this example, so I hope it doesn't scare users and they don't learn it as a best practice. 
However I agree with local `allows`, modify the example as you wish. I recommend to add comment that NUM_CARDS should be <1; 26>.

Reply:
I have added comments and assertions for bad values.